inst/shiny/DiagnosticsExplorer/global.R

```diff
@@ -1,95 +1,80 @@
-defaultPort <- 5432
-defaultUser <- Sys.getenv("shinydbUser")
-defaultPassword <- Sys.getenv("shinydbPw")
-defaultResultsSchema <- 'thrombosisthrombocytopenia'
-defaultVocabularySchema <- defaultResultsSchema
-alternateVocabularySchema <- c('vocabulary')
-
-defaultDatabaseMode <- FALSE # Use file system if FALSE
-
-appInformationText <- "V 2.1.1"
-appInformationText <- "Powered by OHDSI Cohort Diagnostics application - Version 2.1.1 This app is working in"
-if (defaultDatabaseMode) {
-  appInformationText <- paste0(appInformationText, " database")
-} else {
-  appInformationText <- paste0(appInformationText, " local file")
-}
+appInformationText <- "V 2.2.0"
+appInformationText <- "Powered by OHDSI Cohort Diagnostics application - Version 2.2.0"
 appInformationText <- paste0(appInformationText, 
-                             " mode. Application was last initated on ", 
+                             "Application was last initated on ",
                              lubridate::now(tzone = "EST"),
                              " EST. Cohort Diagnostics website is at https://ohdsi.github.io/CohortDiagnostics/")
 
-if (!exists("shinySettings")) {
-  writeLines("Using default settings")
-  databaseMode <- defaultDatabaseMode & defaultServer != ""
-  if (databaseMode) {
-    connectionPool <- pool::dbPool(
-      drv = DatabaseConnector::DatabaseConnectorDriver(),
-      dbms = "postgresql",
-      server = paste(defaultServer, defaultDatabase, sep = "/"),
-      port = defaultPort,
-      user = defaultUser,
-      password = defaultPassword
-    )
-    resultsDatabaseSchema <- defaultResultsSchema
-  } else {
-    dataFolder <- defaultLocalDataFolder
-  }
+
+if (exists("shinySettings")) {
+  writeLines("Using settings provided by user")
+  connectionDetails <- shinySettings$connectionDetails
+  dbms <- connectionDetails$dbms
+  resultsDatabaseSchema <- shinySettings$resultsDatabaseSchema
+  vocabularyDatabaseSchemas <- shinySettings$vocabularyDatabaseSchemas
+} else if (file.exists(sqliteDbPath)){
+  writeLines("Using data directory")
+  sqliteDbPath <- normalizePath(sqliteDbPath)
+  resultsDatabaseSchema <- "main"
+  vocabularyDatabaseSchemas <- "main"
+  dbms <- "sqlite"
+  connectionDetails <- DatabaseConnector::createConnectionDetails(dbms = "sqlite", server = sqliteDbPath)
+} else {
+  writeLines("Connecting to remote database")
+  dbms <- Sys.getenv("shinydbDatabase", unset = "postgresql")
+  connectionDetails <- DatabaseConnector::createConnectionDetails(
+    dbms = dbms,
+    server = Sys.getenv("shinydbServer"),
+    port = Sys.getenv("shinydbPort", unset = 5432),
+    user = Sys.getenv("shinydbUser"),
+    password = Sys.getenv("shinydbPw")
+  )
+
+  resultsDatabaseSchema <- Sys.getenv("shinydbResultsSchema", unset = "thrombosisthrombocytopenia")
+  vocabularyDatabaseSchemas <- resultsDatabaseSchema
+  alternateVocabularySchema <-  Sys.getenv("shinydbVocabularySchema", unset = c("vocabulary"))
```

Comment:
alternateVocabularySchema or referenceVocabulary allowed to compare the impact of vocabulary changes in diagnostics. There can be more than one, and the user gets to toggle between them in the shiny application.

i dont remember if there was support for the the referenceVocabulary in the function that creates shinySettings.

Reply:
`launchDiagnosticsExplorer` has a `vocabularyDatabaseSchemas` parameter so support for multiple vocabs is possible this way